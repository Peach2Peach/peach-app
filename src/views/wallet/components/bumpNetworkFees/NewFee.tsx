import { Dispatch } from "react";
import { View } from "react-native";
import { NumberInput } from "../../../../components/inputs/NumberInput";
import { PeachText } from "../../../../components/text/PeachText";
import { CENT } from "../../../../constants";
import tw from "../../../../styles/tailwind";
import { round } from "../../../../utils/math/round";
import { useTranslate } from "@tolgee/react";

type Props = {
  newFeeRate: string;
  setNewFeeRate: Dispatch<string>;
  overpayingBy?: number;
};

const { t } = useTranslate("wallet");

export const NewFee = ({
  newFeeRate,
  setNewFeeRate,
  overpayingBy = 0,
}: Props) => (
  <View style={tw`items-center gap-10px`}>
    <View
      style={tw`flex-row items-center self-stretch justify-center gap-2 pt-2`}
    >
      <PeachText style={tw`subtitle-1`}>
        {t("wallet.bumpNetworkFees.newFee")}
      </PeachText>
      <View style={tw`h-9`}>
        <NumberInput
          style={tw`w-24 h-9`}
          value={newFeeRate}
          decimals={2}
          placeholder=""
          onChangeText={setNewFeeRate}
          required={true}
        />
      </View>
      <PeachText style={tw`text-center text-black-50`}>
        {t("satPerByte", { ns: "global" })}
      </PeachText>
    </View>
    <PeachText
      style={[
        tw`text-center text-primary-main`,
        overpayingBy < 1 && tw`opacity-0`,
      ]}
    >
      {t("wallet.bumpNetworkFees.overPayingBy", {
        percentage: String(round(overpayingBy * CENT)),
      })}
    </PeachText>
  </View>
);
