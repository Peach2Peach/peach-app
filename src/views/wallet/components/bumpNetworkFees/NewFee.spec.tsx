import ShallowRenderer from 'react-test-renderer/shallow'
import { fireEvent, render } from 'test-utils'
import { NewFee } from './NewFee'

describe('NewFee', () => {
  const setNewFeeRateMock = jest.fn()
  const renderer = ShallowRenderer.createRenderer()
  it('renders correctly', () => {
    renderer.render(<NewFee newFeeRate="1" setNewFeeRate={setNewFeeRateMock} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders correctly when overpaying', () => {
    renderer.render(<NewFee newFeeRate="2" overpayingBy={1} setNewFeeRate={setNewFeeRateMock} />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('sets new fee', () => {
    const newFee = '2'
    const { getByPlaceholderText } = render(<NewFee newFeeRate="1" setNewFeeRate={setNewFeeRateMock} />)
    fireEvent(getByPlaceholderText(''), 'onChangeText', newFee)
    expect(setNewFeeRateMock).toHaveBeenCalledWith(newFee)
  })
})
