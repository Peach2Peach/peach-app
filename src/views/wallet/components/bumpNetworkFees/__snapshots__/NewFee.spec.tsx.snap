// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewFee renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "gap": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "center",
        "paddingTop": 8,
      }
    }
  >
    <PeachText
      style={
        {
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 16,
          "letterSpacing": 0.16,
          "lineHeight": 28,
        }
      }
    >
      new fee:
    </PeachText>
    <View
      style={
        {
          "height": 36,
        }
      }
    >
      <NumberInput
        decimals={2}
        onChangeText={[MockFunction]}
        placeholder=""
        required={true}
        style={
          {
            "height": 36,
            "width": 96,
          }
        }
        value="1"
      />
    </View>
    <PeachText
      style={
        {
          "color": "#9F8C82",
          "textAlign": "center",
        }
      }
    >
      sat/vB
    </PeachText>
  </View>
  <PeachText
    style={
      [
        {
          "color": "#F56522",
          "textAlign": "center",
        },
        {
          "opacity": 0,
        },
      ]
    }
  >
    overpaying by 0%
  </PeachText>
</View>
`;

exports[`NewFee renders correctly when overpaying 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "gap": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "center",
        "paddingTop": 8,
      }
    }
  >
    <PeachText
      style={
        {
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 16,
          "letterSpacing": 0.16,
          "lineHeight": 28,
        }
      }
    >
      new fee:
    </PeachText>
    <View
      style={
        {
          "height": 36,
        }
      }
    >
      <NumberInput
        decimals={2}
        onChangeText={[MockFunction]}
        placeholder=""
        required={true}
        style={
          {
            "height": 36,
            "width": 96,
          }
        }
        value="2"
      />
    </View>
    <PeachText
      style={
        {
          "color": "#9F8C82",
          "textAlign": "center",
        }
      }
    >
      sat/vB
    </PeachText>
  </View>
  <PeachText
    style={
      [
        {
          "color": "#F56522",
          "textAlign": "center",
        },
        false,
      ]
    }
  >
    overpaying by 100%
  </PeachText>
</View>
`;
